### ISSUE TITLE
Investigate performance regressions from BaseDom-initializers PR

### ISSUE THREAD

**Issue by benharsh** [chapel-lang/chapel#9211]
PR #9113 had some nontrivial performance regressions:
[chapcs](https://chapel-lang.org/perf/chapcs/?startdate=2018/03/19&enddate=2018/04/12&graphs=arrayviewcreation,minimdljsize10time,nbodyvariations,reversecomplementshootoutbenchmark)
[16 node xc](https://chapel-lang.org/perf/16-node-xc/?startdate=2018/03/17&enddate=2018/04/11&graphs=prkstencilvariationsperf,prkoptimizedstenciltimesec)

The causes of these regressions are not yet known.

**Comment by benharsh** [comment]
Not sure why, but converting StencilDom to use an explicit initializer appears to resolve the PRK Stencil regression. Odd...

**Comment by benharsh** [comment]
Also not seeing any increased communication when comparing before/after the BaseDom PR for PRK Stencil.

**Comment by benharsh** [comment]
Hmm, even weirder, it's possible to get the bad performance in the explicit initializer if one calls this.complete() and assigns to locDoms:
```chpl
this.complete();
var foo : [dist.targetLocDom] LocStencilDom(rank, idxType, stridable);
this.locDoms = foo;
```
This is what the default initializer does, sorta.

Doesn't make sense to me that this would result in a performance loss. Creation and privatization of StencilDom is all done before the timed section, and the locDoms array and the default/temp should have the same locality, numa-affinity, and values (nil).

**Comment by benharsh** [comment]
Applying the task-resetting fix from #7831 resolves the PRK Stencil regression. For now I will work around this by adding an explicit initializer to StencilDom.

**Comment by benharsh** [comment]
The other regressing tests can be fixed by adding an explicit initializer to ``BaseDom``, which avoids unnecessary default arg creation. I suspect the "_arrsLock" field to be the problematic one.
